### PR TITLE
feat(compute): attach/ remove snapshot schedule to disk

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Documentation - https://cloud.google.com/compute/docs/disks/scheduled-snapshots?authuser=0#attach_snapshot_schedule
Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/snapshotschedule/AttachSnapshotScheduleToDisk.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  // Attaches a snapshot schedule to a disk.
+  public static Status attachSnapshotScheduleToDisk(
+        String projectId, String zone, String diskName, String snapshotScheduleName, String region)
+        throws IOException, ExecutionException, InterruptedException, TimeoutException {
+
+    String resourcePolicyLink = String.format(
+            "projects/%s/regions/%s/resourcePolicies/%s", projectId, region, snapshotScheduleName);
+    try (DisksClient disksClient = DisksClient.create()) {
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/AttachSnapshotScheduleToDisk.java:54]: Thank you! Added comment

File: compute/cloud-client/src/main/java/compute/snapshotschedule/AttachSnapshotScheduleToDisk.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+              .setDisk(diskName)
+              .setDisksAddResourcePoliciesRequestResource(
+                      DisksAddResourcePoliciesRequest.newBuilder()
+                              .addResourcePolicies(resourcePolicyLink)
+                              .build())
+              .setProject(projectId)
+              .setZone(zone)
+              .build();
+
+      Operation response = disksClient.addResourcePoliciesAsync(request).get(1, TimeUnit.MINUTES);
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/AttachSnapshotScheduleToDisk.java:66]: Thank you! Fixed timeout to 3 min

File: compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.snapshotschedule;
+
+// [START compute_snapshot_schedule_create]
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/CreateSnapshotSchedule.java:19]: I need this classes for testing.

File: compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.snapshotschedule;
+
+// [START compute_snapshot_schedule_delete]
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/DeleteSnapshotSchedule.java:19]: I need this classes for testing.

File: compute/cloud-client/src/main/java/compute/snapshotschedule/RemoveSnapshotScheduleFromDisk.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the zone where your disk is located.
+    String zone = "us-central1-a";
+    // Name of the disk you want to remove the snapshot schedule from.
+    String diskName = "YOUR_DISK_NAME";
+    // Name of the region where your snapshot schedule is located.
+    String region = "us-central1";
+    // Name of the snapshot schedule you want to attach.
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/RemoveSnapshotScheduleFromDisk.java:42]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/snapshotschedule/RemoveSnapshotScheduleFromDisk.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+              DisksRemoveResourcePoliciesRequest.newBuilder()
+                      .addResourcePolicies(snapshotSchedulePath)
+                      .build();
+
+      RemoveResourcePoliciesDiskRequest request =
+              RemoveResourcePoliciesDiskRequest.newBuilder()
+                      .setDisk(diskName)
+                      .setProject(project)
+                      .setDisksRemoveResourcePoliciesRequestResource(disksRequest)
+                      .setZone(zone)
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/snapshotschedule/RemoveSnapshotScheduleFromDisk.java:68]: Thank you! Fixed

File: compute/cloud-client/src/test/java/compute/snapshotschedule/SnapshotScheduleIT.java [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+  @Test
+  @Order(1)
+  public void testCreateSnapshotScheduleHourly()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Status status = CreateSnapshotSchedule.createSnapshotSchedule(
+            PROJECT_ID, REGION, SCHEDULE_NAME, SCHEDULE_DESCRIPTION,
+            MAX_RETENTION_DAYS, STORAGE_LOCATION, ON_SOURCE_DISK_DELETE);
+
+    assertThat(status).isEqualTo(Status.DONE);
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/snapshotschedule/SnapshotScheduleIT.java]: Removed not related tests and used a mock client to test code.

**Review by Sita04 — APPROVED** [review]

**Pull request merged**